### PR TITLE
Feature hor 3 edit calendar

### PR DESCRIPTION
**Pull request by asfram**
This PR adds an edit action on a calendar in the calendars list

http://jira.canaltp.fr/browse/HOR-3
This is a  new feature

Team Schedule

### CODE REVIEW

File: Controller/CalendarController.php [hunk 1]
```diff
@@ -97,4 +97,32 @@ public function listAction()
...
+    public function editAction(Request $request, $calendarId)
+    {
+        $em = $this->get('doctrine')->getManager();
+        $translator = $this->get('translator');
+
+        $calendar = $this->getDoctrine()
+            ->getRepository('CanalTPMttBundle:Calendar')
+            ->findOneBy(
+                ['customer' => $this->getUser()->getCustomer(),'id' => $calendarId]
+         );
```
Comment (datanel) [inline Controller/CalendarController.php]: what happens if $calendarId does not exist?

**Comment by datanel** [comment]
FILE: ...tt-bundle/Tests/Functional/Controller/CalendarControllerTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 62 | ERROR | [x] Opening brace should be on a new line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------


FILE: ...p.fr/vendor/canaltp/mtt-bundle/Controller/CalendarController.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------
 110 | ERROR | [x] Multi-line function call not indented correctly;
     |       |     expected 12 spaces but found 9
 110 | ERROR | [x] Line indented incorrectly; expected at least 12
     |       |     spaces, found 9
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 7.27 secs; Memory: 31.25Mb

File: Controller/CalendarController.php [hunk 2]
```diff
@@ -97,4 +97,32 @@ public function listAction()
...
+
+        $form = $this->createForm(new CalendarType(), $calendar);
+        $form->add('submit', 'submit', ['label' => 'global.validate', 'translation_domain' => 'messages']);
+
+        $form->handleRequest($request);
+
+        if ($request->isMethod('POST') && $form->isValid()) {
+            $calendar->setCustomer($this->getUser()->getCustomer());
```
Comment (datanel) [inline Controller/CalendarController.php]: this line is useless as you already have the customer from the query above

File: Tests/Functional/Controller/CalendarControllerTest.php [hunk 3]
```diff
@@ -55,6 +55,119 @@ public function testCalendarsCreateAction()
...
+
+        // Assert that edit button exists for a calendar
+        $calendarsFirstRowEditButton = $calendarsFirstRow->filter('td a[href*="edit"]');
+        $this->assertCount(1, $calendarsFirstRowEditButton);
+
+        // Assert that edit button redirect to calendar edit page
+        $calendarsFirstRowEditLink = $calendarsFirstRowEditButton->link();
+        $crawler = $this->client->click($calendarsFirstRowEditLink);
+        $expectedUri = $calendarsFirstRowEditLink->getUri();
+        $this->assertEquals($calendarsFirstRowEditLink->getUri(), $this->client->getRequest()->getUri());
```
Comment (datanel) [inline Tests/Functional/Controller/CalendarControllerTest.php]: not sure of this test

you click on a link
and you test uri = link

it seems normal

you should test link = '/mtt/edit/6' (for example) before clicking

File: Controller/CalendarController.php (hunk 1 above)
Comment (asfram) [inline Controller/CalendarController.php]: DONE

File: Controller/CalendarController.php (hunk 2 above)
Comment (asfram) [inline Controller/CalendarController.php]: DONE

File: Tests/Functional/Controller/CalendarControllerTest.php (hunk 3 above)
Comment (asfram) [inline Tests/Functional/Controller/CalendarControllerTest.php]: DONE

File: Tests/Functional/Controller/CalendarControllerTest.php [hunk 4]
```diff
@@ -55,6 +55,128 @@ public function testCalendarsCreateAction()
         $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
         $this->assertCount(1, $crawler->filter('html:contains("Le calendrier a été créé")'));
     }
+    
+    /**
+     *  @group editCalendar
```
Comment (datanel) [inline Tests/Functional/Controller/CalendarControllerTest.php]: you can remove this group

Comment (asfram) [inline Tests/Functional/Controller/CalendarControllerTest.php]: DONE

**Comment by datanel** [comment]
PR OK but we need to fix jenkins

**Pull request merged**